# HG changeset patch
# User prr
# Date 1449082041 28800
#      Wed Dec 02 10:47:21 2015 -0800
# Branch zulu8-marlin
# Node ID 6aa4a7ea1682a49a967175b29df751322a0d86f9
# Parent  52a6785e245aef94598b01fc6433ffd7b225f6a8
8144526: Remove Marlin logging use of deleted internal API
Reviewed-by: flar

diff --git a/src/share/classes/sun/java2d/marlin/MarlinUtils.java b/src/share/classes/sun/java2d/marlin/MarlinUtils.java
--- a/src/share/classes/sun/java2d/marlin/MarlinUtils.java
+++ b/src/share/classes/sun/java2d/marlin/MarlinUtils.java
@@ -25,8 +25,6 @@
 
 package sun.java2d.marlin;
 
-import sun.misc.JavaLangAccess;
-import sun.misc.SharedSecrets;
 
 public final class MarlinUtils {
     // TODO: use sun.util.logging.PlatformLogger once in JDK9
@@ -70,31 +68,6 @@
         String sourceClassName = null;
         String sourceMethodName = null;
 
-        JavaLangAccess access = SharedSecrets.getJavaLangAccess();
-        Throwable throwable = new Throwable();
-        int depth = access.getStackTraceDepth(throwable);
-
-        boolean lookingForClassName = true;
-        for (int ix = 0; ix < depth; ix++) {
-            // Calling getStackTraceElement directly prevents the VM
-            // from paying the cost of building the entire stack frame.
-            StackTraceElement frame = access.getStackTraceElement(throwable, ix);
-            String cname = frame.getClassName();
-            if (lookingForClassName) {
-                // Skip all frames until we have found the first frame having the class name.
-                if (cname.equals(className)) {
-                    lookingForClassName = false;
-                }
-            } else {
-                if (!cname.equals(className)) {
-                    // We've found the relevant frame.
-                    sourceClassName = cname;
-                    sourceMethodName = frame.getMethodName();
-                    break;
-                }
-            }
-        }
-
         if (sourceClassName != null) {
             return sourceClassName + " " + sourceMethodName;
         } else {
